### PR TITLE
Faeries/multiple choice question

### LINKED ISSUE #151: questionbotting / add ask_multiplechoice_question
```
checker - sees if answer can be converted
converter - converts string to type of answer
clarifier - text added to question if checker fails
```

``` python

ask_multiple_choice ( recipient, question_text: Optional[str], options: Union[dict[str,str], list[str]], requires_confirmation: bool, requires_first_device: bool):

if question_text is None:
    question_text = "Pick one from these options:"
if isinstance(options, dict):
   # labels provided
if isinstance(options, list):
   # generate labels

options = dict[str,str]

prints ie)

"""Please enter one of the following numbers to select a charity to receive your donation:
    1. Ideas Beyond Borders ( https://www.ideasbeyondborders.org/ )
    2. Ukrainian Students for Freedom ( https://standwithukraine.ch/ )"""
```

``` python
{'give': 'Please enter one of the following numbers to select a charity to receive your donation:\n    1. Ideas Beyond Borders ( https://www.ideasbeyondborders.org/ )\n    2. Ukrainian Students for Freedom ( https://standwithukraine.ch/ )', 'gift': "You probably mean 'give'! - updated by ilia_76ae", 'donate': "You probably mean 'give'! - added by ilia_76ae", 'longnow': 'The Long Now Foundation', 'ideasbeyondborders': 'Ideas Beyond Borders (https://www.ideasbeyondborders.org/)', 'fightforthefuture': 'Fight For The Future', '1_give': 'ideasbeyondborders', '2_give': 'standwithukraine', '3_give': 'standwithukraine', 'comebackalive': 'Повернись Живим (Come Back Alive)', 'standwithukraine': 'Ukranian Students for Freedom (https://standwithukraine.ch/)', '1': '"Say \'give\' first please!" - added by ilia_76ae', '2': "Say 'give' first please! - added by ilia_76ae", '3': "Say 'give' first please! - added by ilia_76ae", '4': "Sorry, that's not an option! - added by ilia_76ae"}
```

### PR DESCRIPTION
**Pull request by transfaeries**
Closes #151 

Adds an ask_multiple_choice_question function to questionbot core. Currently it allows for 3 behaviours:

if user provides options as a list, print that list with numbers, accepts exact numbers as answers:
eg:
What is your favourite forest creature?
1) Deer 
2) Foxes 
3) Faeries 
4) Crows 

if user provides options as a dict, we use the keys as the labels and accept the exact label
eg:
What is your favourite forest creature?
A) Deer 
B) Foxes 
⛧) Faeries 
D) Crows 

if user provides options as a dict but with empty values, we use only the keys and again, match exactly
eg:
What is your tshirt size?
S 
M 
L 
XL 
XXL 

Requires_confirmation defaults to true and asks users to confirm their choice with a yes or no question.

Possible improvements would be to allow fuzzy matching, possibly behind a flag, that would allow for things like matching user input to values instead of keys, or allowing lowercase answers

Additionally: 
tests/test_questionbot.py defines a number of test functions, that just test the basic functionality:
test_ask_yesno_question
test_multiple_choice_dict_emptyval
test_multiple_choice_dict_no_confirm
test_multiple_choice_dict_with_confirm
test_multiple_choice_list_no_confirm
test_multiple_choice_list_with_confirm

### CODE REVIEW

**Comment by transfaeries** [comment]
```
forest/core.py:1305:4: R0913: Too many arguments (6/5) (too-many-arguments)
forest/core.py:1305:4: R0914: Too many local variables (17/15) (too-many-locals)
```
Not actually sure how to fix these. Like, why are these the limits.

File: forest/core.py [hunk 1]
```diff
@@ -1298,6 +1302,112 @@ async def ask_yesno_question(
         self.pending_confirmations.pop(recipient)
         return result
 
+    async def ask_multiple_choice_question(
+        self,
+        recipient: str,
+        question_text: Optional[str],
+        options: Union[dict[str, str], list[str]],
+        requires_confirmation: bool = True,
+        requires_first_device: bool = False,
```
[the remark this answers is not in the document]
Comment (transfaeries) [reply forest/core.py:1311]: where do I put 
`pylint: disable=too-many-arguments`?

**Comment by transfaeries** [comment]
f"{label}{spacer if body else ''}{body}"
            for label, body in dict_options.items()
            
 doesn't work well cause I want it to be blank only if all items are blank. And I need the spacer to be blank because later on we use it for confirmation.

is there a pretty way to keep that functionality?

**Comment by transfaeries** [comment]
I actually just found a bug, give me a minute to fix it

**Review by technillogue — APPROVED** [review]

**Pull request merged**